pkg/api/types.go

```diff
@@ -317,13 +323,11 @@ type Container struct {
 	WorkingDir string   `json:"workingDir,omitempty"`
 	Ports      []Port   `json:"ports,omitempty"`
 	Env        []EnvVar `json:"env,omitempty"`
-	// Optional: Defaults to unlimited.
-	Memory resource.Quantity `json:"memory,omitempty"`
-	// Optional: Defaults to unlimited.
-	CPU           resource.Quantity `json:"cpu,omitempty"`
-	VolumeMounts  []VolumeMount     `json:"volumeMounts,omitempty"`
-	LivenessProbe *Probe            `json:"livenessProbe,omitempty"`
-	Lifecycle     *Lifecycle        `json:"lifecycle,omitempty"`
+	// Compute resource requirements.
+	Resources     ResourceRequirementSpec `json:"resources,omitempty"`
```

Comment:
nit: We don't say "Spec" on any other types except the ones that correspond to a "spec" field.